compiler/tests/generated/alias_with_allocator_types.h

```diff
@@ -21,6 +21,8 @@ namespace test
     
     struct foo
     {
+        using allocator_type = arena;
+
         std::list<bool, typename std::allocator_traits<arena>::template rebind_alloc<bool> > l;
```

Comment:
We can now use `allocator_type` in all such places instead of using the original name (`arena` in this case). Let me know what you think.

Reply:
Seems cleaner to not repeat the name all over the place. It will churn the golden files more, but that's OK.

Reply:
Will send a separate PR for that.

Reply:
This won't work, since we need to use the original name for a generic base type of string/container (e.g. `struct Derived : Base<string>`).